itests/fevm_test.go

```diff
@@ -786,3 +788,57 @@ func TestFEVMDestroyCreate2(t *testing.T) {
 	require.Equal(t, ethFromAddr, senderSecondCall)
 
 }
+
+func TestFEVMBareTransferTriggersSmartContractLogic(t *testing.T) {
+	ctx, cancel, client := kit.SetupFEVMTest(t)
+	defer cancel()
+
+	// This contract emits an event on receiving value.
+	filename := "contracts/ValueSender.hex"
+	_, contractAddr := client.EVM().DeployContractFromFilename(ctx, filename)
+
+	accctKey, accntEth, accntFil := client.EVM().NewAccount()
+	kit.SendFunds(ctx, t, client, accntFil, types.FromFil(10))
+
+	contractEth, err := ethtypes.EthAddressFromFilecoinAddress(contractAddr)
+	require.NoError(t, err)
+
+	gaslimit, err := client.EthEstimateGas(ctx, ethtypes.EthCall{
+		From:  &accntEth,
+		To:    &contractEth,
+		Value: ethtypes.EthBigInt(types.FromFil(1)),
+	})
+	require.NoError(t, err)
+
+	maxPriorityFeePerGas, err := client.EthMaxPriorityFeePerGas(ctx)
+	require.NoError(t, err)
```

Comment:
This logic is repeated in various places and it's worth extracting to a helper in the itests kit, which takes an *EthTxArgs, forms the EthCall, estimates gas, gets the priority fee, and fills those details in the Eth tx.